Source/CarthageKit/Project.swift

```diff
@@ -229,27 +237,33 @@ public final class Project {
 
 	/// Runs the given Git operation, blocking the `gitOperationScheduler` until
 	/// it has completed.
-	private func runGitOperation<T>(operation: ColdSignal<T>) -> ColdSignal<T> {
-		return ColdSignal { (sink, disposable) in
-			let schedulerDisposable = self.gitOperationScheduler.schedule {
-				let results = operation
-					.reduce(initial: []) { $0 + [ $1 ] }
-					.first()
-
-				switch results {
-				case let .Success(values):
-					ColdSignal.fromValues(values.unbox).startWithSink { valuesDisposable in
-						disposable.addDisposable(valuesDisposable)
-						return sink
+	private func runGitOperation<T, Error>(operation: SignalProducer<T, Error>) -> SignalProducer<T, Error> {
+		return SignalProducer { observer, disposable in
+				let schedulerDisposable = self.gitOperationScheduler.schedule {
+					let results = operation
+						|> reduce([]) { $0 + [ $1 ] }
```

Comment:
Good point, but I've generally avoided “semantic” changes. I just want to make this a source-to-source translation as much as possible.